class.jetpack.php

```diff
@@ -1491,6 +1494,35 @@ public static function is_development_mode() {
 	}
 
 	/**
+	 * Filter enqueued script and minify Jetpack's ones
+	 */
+	public function load_minified_scripts( $tag, $handle, $src ) {
+		$jetpack_url = plugins_url( '', __FILE__ );
+
+		$load_minified_scripts = ! ( ( defined( 'SCRIPT_DEBUG' ) && SCRIPT_DEBUG ) ||
+		                             Jetpack::is_development_mode() || ( defined( 'IS_WPCOM' ) && IS_WPCOM ) );
+
+		/**
+		 * Filters Jetpack's development mode.
+		 *
+		 * @since 4.0.0
+		 *
+		 * @param bool $development_mode Whether to load minified script.
+		 */
+		$load_minified_scripts = apply_filters( 'jetpack_load_minified_scripts', $load_minified_scripts );
+
+		if ( false !== strpos( $src, $jetpack_url ) && $load_minified_scripts ) {
```

Comment:
This block feels pretty brittle, and prone to problems (especially if there are files along the lines of `core.jsdom.js` or something that has `.js` in amongst it.

You might need to parse the URLs a bit deeper and look at the `basename`, then regex or whatever to see that specifically the last 3 characters are `.js`.